sbi/utils/get_nn_models.py

```diff
@@ -2,39 +2,155 @@
 # under the Affero General Public License v3, see <https://www.gnu.org/licenses/>.
 
 
-from typing import Callable, Optional, Union, Dict, Any, Tuple, Union, cast, List, Sequence, TypeVar
-from sbi.neural_nets.flow import build_made, build_maf, build_nsf
-from sbi.neural_nets.mdn import build_mdn
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from torch import nn
+
+import sbi
 from sbi.neural_nets.classifier import (
     build_linear_classifier,
     build_mlp_classifier,
     build_resnet_classifier,
 )
+from sbi.neural_nets.flow import build_made, build_maf, build_nsf
+from sbi.neural_nets.mdn import build_mdn
+
+
+def classifier_nn(
+    model: str,
+    z_score_theta: bool = True,
+    z_score_x: bool = True,
+    hidden_features: int = 50,
+    embedding_net_theta: nn.Module = nn.Identity(),
+    embedding_net_x: nn.Module = nn.Identity(),
+) -> Callable:
+    r"""
+    Returns a function that builds a classifier for learning density ratios.
+
+    This function will usually be used for SNRE. The returned function is to be passed
+    to the inference class when using the flexible interface.
+
+    Args:
+        model: The type of classifier that will be created. One of [`linear`, `mlp`,
+            `resnet`].
+        z_score_theta: Whether to z-score parameters $\theta$ before passing them into
+            the network.
+        z_score_x: Whether to z-score simulation outputs $x$ before passing them into
+            the network.
+        hidden_features: Number of hidden features.
+        embedding_net_theta:  Optional embedding network for parameters $\theta$.
+        embedding_net_x:  Optional embedding network for simulation outputs $x. This
+            embedding net allows to learn features from potentially high-dimensional
+            simulation outputs.
+    """
 
+    kwargs = dict(
+        zip(
+            (
+                "z_score_x",
+                "z_score_y",
+                "hidden_features",
+                "embedding_net_x",
+                "embedding_net_y",
+            ),
+            (
+                z_score_x,
+                z_score_theta,
+                hidden_features,
+                embedding_net_x,
+                embedding_net_theta,
+            ),
+        )
+    )
 
-def classifier_nn(model: str, **kwargs: Any) -> Callable:
     def build_fn(batch_theta, batch_x):
         if model == "linear":
+            linear_kwargs = sbi.utils.del_entries(kwargs, ("hidden_features",))
```

Comment:
So we are settings default values for all possible kwargs for all possible nn types and then for the specific types we delete the unneeded kwargs from the dict?

Reply:
yes, exactly. This is the cleanest option I could think of. Let me know if you have better ideas though, I'm not 100% happy with it either.

Reply:
Yes, I think it is transparent way for now. As always, there might be better ways of doing it, e.g., when we start using [thinc](https://github.com/explosion/thinc) at one point.